inc/shortcodes/class-image-comparision.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class Image_Comparision extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => 'Image Comparison',
+			'listItemImage'  => 'dashicons-format-gallery',
+			'attrs'          => array(
+				array(
+					'label'  => 'Left Image',
+					'attr'   => 'left',
+					'type'   => 'attachment',
+					'libraryType' => array( 'image' ),
+					'addButton'   => 'Select Image',
+					'frameTitle'  => 'Select Image',
+					),
+				array(
+					'label'  => 'Right Image',
+					'attr'   => 'right',
+					'type'   => 'attachment',
+					'libraryType' => array( 'image' ),
+					'addButton'   => 'Select Image',
+					'frameTitle'  => 'Select Image',
+					),
+				array(
+					'label'  => 'Slider Start Position',
+					'attr'   => 'position',
+					'type'   => 'select',
+					'options' => array(
+						'center' => 'Center',
+						'mostlyleft' => 'Mostly Left',
+						'mostlyright' => 'Mostly Right'
+						)
+					),
+				),
+		);
+	}
+
+	public static function setup_actions() {
+		add_action( 'init', 'Shortcake_Bakery\Shortcodes\Image_Comparision::action_init_register_scripts' );
+		add_action( 'init', 'Shortcake_Bakery\Shortcodes\Image_Comparision::action_init_register_styles' );
+	}
+
+	public static function action_init_register_scripts() {
+		wp_register_script( 'juxtapose-js', SHORTCAKE_BAKERY_URL_ROOT . 'assets/js/juxtapose.js', array( 'jquery' ) );
+	}
+
+	public static function action_init_register_styles() {
+		wp_register_style( 'juxtapose-css', SHORTCAKE_BAKERY_URL_ROOT . 'assets/css/juxtapose.css' );
+	}
+
+	public static function callback( $attrs, $content = '' ) {
+		if ( empty( $attrs['left'] ) || empty( $attrs['right'] ) ) {
+			if ( current_user_can( 'edit_posts' ) ) {
+				return '<div class="shortcake-bakery-error"><p>' . esc_html__( 'Two images required for image comparision.', 'shortcake-bakery' ) . '</p></div>';
+			} else {
+				return '';
+			}
+		}
+
+		if ( !isset( $attrs['position'] ) ) {
+			 $attrs['position'] = 'center';
+		}
+
+		switch ( $attrs['position'] ) {
+
+			case 'center' :
+				$attrs['position'] = 50;
+				break;
+
+			case 'mostlyleft' :
+				$attrs['position'] = 10;
+				break;
+
+			case 'mostlyright' :
+				$attrs['position'] = 90;
+				break;
+
+		}
+
+		$left_image = wp_get_attachment_image_src( $attrs['left'], 'large', false );
+		$right_image = wp_get_attachment_image_src( $attrs['right'], 'large', false );
+
+		$left_caption = get_post_field( 'post_excerpt', $attrs['left'] );
+		$right_caption = get_post_field( 'post_excerpt', $attrs['right'] );
+
+		if ( ! $left_image || ! $right_image ) {
+			return;
+		}
+
+		wp_enqueue_script( 'juxtapose-js' );
```

Comment:
@danielbachhuber `wp_enqueue_scripts` doesn't seem to run within the admin view.